src/dsymbol.d

```diff
@@ -460,20 +460,20 @@ extern (C++) class Dsymbol : RootObject
     // instantiated speculatively, eg from inside is(typeof()).
     // Return the speculative template instance it is part of,
     // or NULL if not speculative.
-    final TemplateInstance isSpeculative()
+    final inout(TemplateInstance) isSpeculative() inout
     {
-        Dsymbol par = parent;
+        Dsymbol par = cast()parent;
```

Comment:
Sorry, I don't know how I managed to overlook this.

Perhaps Rebindable could be used to encapsulate tail-only const and to make the code `@safe`. Anyway it looks good for now.